File: academic-system/src/main/java/org/example/repository/RepositorioUsuarioTxt.java
package org.example.repository;

import org.example.model.Usuario;
import java.io.BufferedReader;
import java.io.FileReader;
import java.util.Optional;

public class RepositorioUsuarioTxt implements RepositorioUsuario {
    private final String caminhoArquivo;

    public RepositorioUsuarioTxt(String caminhoArquivo) {
        this.caminhoArquivo = caminhoArquivo;
    }

    @Override
    public Optional<Usuario> buscarPorUsername(String username) {
        try (BufferedReader reader = new BufferedReader(new FileReader(caminhoArquivo))) {
            String linha;
            while ((linha = reader.readLine()) != null) {
                String[] partes = linha.split(",");
                if (partes.length >= 3 && partes[0].equals(username)) {
                    return Optional.of(new Usuario(partes[0], partes[1], partes[2]));
                }
            }
        } catch (Exception e) {
            // Silencioso, retorna vazio
        }
        return Optional.empty();
    }
}

File: academic-system-csharp/SistemaAcademico/src/SistemaAcademico/Repositories/RepositorioUsuarioTxt.cs
using SistemaAcademico.Models;

namespace SistemaAcademico.Repositories;

// Equivalente à classe Java: RepositorioUsuarioTxt implements RepositorioUsuario
// java.io.BufferedReader / FileReader → System.IO.StreamReader
// Optional<Usuario> → Usuario? (nullable reference)
// Comportamento idêntico: ler CSV linha a linha, split por vírgula, retornar usuário ou null
public class RepositorioUsuarioTxt : IRepositorioUsuario
{
    private readonly string caminhoArquivo;

    public RepositorioUsuarioTxt(string caminhoArquivo)
    {
        this.caminhoArquivo = caminhoArquivo;
    }

    public Usuario? BuscarPorUsername(string username)
    {
        try
        {
            using StreamReader reader = new StreamReader(caminhoArquivo);
            string? linha;
            while ((linha = reader.ReadLine()) != null)
            {
                string[] partes = linha.Split(',');
                if (partes.Length >= 3 && partes[0].Equals(username))
                {
                    return new Usuario(partes[0], partes[1], partes[2]);
                }
            }
        }
        catch (Exception)
        {
            // Silencioso, retorna null (equivalente ao Optional.empty())
        }
        return null;
    }
}
